src/sentry/integrations/slack/notifications.py

```diff
@@ -43,15 +43,23 @@ def get_channel_and_integration_by_user(
         # recipients.
         return None, None
 
+    identity = [
+        identity
+        for identity in identities
+        if identity.user.org_memberships.filter(id=organization.id)
```

Comment:
is there some cool way to add this filter to the db query? I couldn't figure out how to do it

Reply:
I'm not sure this actually helps. Every identity here is already filtered to belong to one specific user, and I assume that we already know the user is a member of the org?